base_layer/core/src/transactions/key_manager/inner.rs

```diff
@@ -1003,26 +1070,92 @@ where TBackend: KeyManagerBackend<PublicKey> + 'static
 
                 #[cfg(feature = "ledger")]
                 {
-                    let mut total_script_private_key = PrivateKey::default();
+                    let mut partial_script_offset = PrivateKey::default();
+                    let mut derived_script_keys = vec![];
+                    let mut script_key_indexes = vec![];
+                    for script_key_id in script_key_ids {
+                        match script_key_id {
+                            TariKeyId::Managed { branch, index } => {
+                                match TransactionKeyManagerBranch::from_key(branch) {
+                                    TransactionKeyManagerBranch::Spend | TransactionKeyManagerBranch::PreMine => {
+                                        script_key_indexes
+                                            .push((TransactionKeyManagerBranch::from_key(branch), *index));
+                                    },
+                                    _ => {
+                                        return Err(TransactionError::from(
+                                            self.branch_not_supported_error("get_script_offset", branch),
+                                        ));
+                                    },
```

Comment:
```suggestion
                                    _ => {
                                        partial_script_offset += self.get_private_key(script_key_id).await? 
                                    },
```
This is incorrect. Keymanager should not force key types used. This should be up to the higher level logic